client/src/plugins/update-checks/UpdateChecks.js

```diff
@@ -74,6 +74,7 @@ export default class UpdateChecks extends PureComponent {
       const updateCheckInfo = await config.get(UPDATE_CHECKS_CONFIG_KEY) || {};
 
       updateCheckInfo.stagedRollout = false;
+      updateCheckInfo.latestVersion = 'v' + Metadata.data.version;
```

Comment:
Should we monkey-patch here? To be safe, let us not do that, i.e. do the following instead:

```
const updateCheckInfo = {
  ...loadedInfo
  stagedRollout: false,
  latestVersion: 'v' + Metadata.data.version
};
```

Reply:
I know the code _may work_ but I would still submit it to our hour of code if I found it like this (including the `stagedRollout` thing).

Reply:
Additionally: would this not still potentially cause this bug described here: https://github.com/camunda/camunda-modeler/issues/2402#issuecomment-906416494 ?

I.e. the bug is IMO, that the version retrieved is catched from the config.json, and not from the actual running Camunda Modeler version.

Reply:
> Additionally: would this not still potentially cause this bug described here: [#2402 (comment)](https://github.com/camunda/camunda-modeler/issues/2402#issuecomment-906416494) ?
> 
> I.e. the bug is IMO, that the version retrieved is catched from the config.json, and not from the actual running Camunda Modeler version.

What gets populated into `newerThan` is actually the `latestVersion`. Every time a version is skipped `latestVersion` is increased, and obviously when we install an old version of the app, we end up having the equivalent of skipped versions.

So what my initial fix would have done is reset `latestVersion` on manual update, which would fix the problem. The proposed alternative by Niko does the same thing but does not modify the cached value.

I am also wondering why `stagedRollout` is even cached in the first place. There does not need to be a use case for this in the codebase, it is always just set in all its execution paths unless I missed something.

Reply:
Thanks for explaining - got it now. 

Since the cached value is used for the scheduled check, I think it makes sense to conceptually not modify it?